server/api/authentication/auth_handler.go

```diff
@@ -0,0 +1,80 @@
+package authentication
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"github.com/tobocop/go-teleport-directory-browser/api/session"
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type AuthRequest struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+func (s *Server) AuthHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	var authReq AuthRequest
+	err := json.NewDecoder(req.Body).Decode(&authReq)
+	if err != nil {
+		log.Printf("AuthHandler json decode error: %v", err)
+		w.WriteHeader(400)
```

Comment:
1) Use of a constant would be preferred here, so `http.StatusBadRequest` instead of `400`.

2) This suggestion is minor, you don't need to make this change, but I usually use `http.Error` for this because it will set the response body with a human readable message as well.

```go
http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
```

Reply:
Good call on constants. Not sure why I was so sporadic on their usage in this file and the tests. 

I didn't know about `http.Error()`, that's pretty nice

Reply:
Just ran a test, JSON parser catches it so no need to encode / decode things.